--- cmd/aws-lambda-rie/http.go	2022-05-06 18:02:41.000000000 -0700
+++ cmd/aws-lambda-rie/http-new.go	2022-05-06 18:02:19.000000000 -0700
@@ -6,21 +6,21 @@
 import (
 	"net/http"

+	"github.com/rs/cors"
+
 	log "github.com/sirupsen/logrus"
 )

 func startHTTPServer(ipport string, sandbox Sandbox) {
-	srv := &http.Server{
-		Addr: ipport,
-	}
-
 	// Pass a channel
-	http.HandleFunc("/2015-03-31/functions/function/invocations", func(w http.ResponseWriter, r *http.Request) {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/2015-03-31/functions/function/invocations", func(w http.ResponseWriter, r *http.Request) {
 		InvokeHandler(w, r, sandbox)
 	})

 	// go routine (main thread waits)
-	if err := srv.ListenAndServe(); err != nil {
+	handler := cors.Default().Handler(mux)
+	if err := http.ListenAndServe(ipport, handler); err != nil {
 		log.Panic(err)
 	}

